clang/utils/TableGen/ClangProgModelBuiltinEmitter.cpp

```diff
@@ -848,6 +916,232 @@ void BuiltinNameEmitter::EmitQualTypeFinder() {
   OS << "\n} // Bultin2Qual\n";
 }
 
+JSONBuiltinInterfaceEmitter::TypeDesc::TypeDesc(const Record *T)
+    : Name(T->getValueAsString("Name")), VecWidth(T->getValueAsInt("VecWidth")),
+      ElementSize(T->getValueAsInt("ElementSize")),
+      IsPointer(T->getValueAsBit("IsPointer")),
+      IsConst(T->getValueAsBit("IsConst")),
+      IsVolatile(T->getValueAsBit("IsVolatile")) {
+  assert((!T->getValueAsBit("IsInteger") || !T->getValueAsBit("IsFloat")) &&
```

Comment:
Can you please add the isSubClassOf checks as well?  It seems that it shouldn't be a BUILTIN either.

Reply:
It is already handled by the unreachable.

Reply:
I don't think it is.  I'm talking about a type that IsInteger, IsFloat, and isSubClassOf(FundamentalType) AND isSubClassOf(VectorType).

At the moment you catch only isInteger and isFloat, but if you leave any of those off it isnt. 

Additionally, the logic below allows a type to be both a fundamental and vector type.